docs/app-integration-development/08-OAuth-2-Functionality.md

```diff
@@ -51,5 +51,187 @@ There two options for implementing PagerDuty OAuth in your app. [PKCE (Proof Key
+ `purpose`       |  Purpose of the JWT, which in an ID token's case is always identification
+ `at_hash`       |  ID Token hash value
+ `acr`           |  The authentication context class reference
+ `delegation_id` |  ID Token delegation that tracks permissions of OAuth clients
+ `account_id`    |  ID of the account linked to the token
+ `user_id`       |  ID of the user linked to the token 
+ `azp`           |  The authorized party that was issued the ID Token, which is the client ID
+ `amr`           |  Authentication method references
+ `subdomain`     |  PagerDuty subdomain linked to the ID token 
+ `region`        |  User and account service region
+ `sid`           |  Session ID
+
+Putting all these fields together would give us a token with the following fields:
+
+```json
+{
+  "exp": 1646086260,
+  "nbf": 1646082660,
+  "jti": "a9fd43a4-b037-4eeb-98b0-40542e3b98ff",
+  "iss": "https://app.pagerduty.com/global/oauth/anonymous",
+  "aud": [
+    "https://api.pagerduty.com",
+    "90b153bf-cfeb-4837-8b6a-864b7ff85656"
+  ],
+  "sub": "example@pagerduty.com",
+  "auth_time": 1646082649,
+  "iat": 1646082660,
+  "purpose": "id",
+  "at_hash": "LMtFKMLYeFeziNtx9PJWHA",
+  "acr": "acr:html-form:unitedstates",
+  "delegation_id": "6566027d-7707-428f-8752-3ef971b78d1a",
+  "account_id": "P9JEHMK",
+  "user_id": "P8OGMIK",
+  "azp": "90b153bf-cfeb-4837-8b6a-864b7ff85656",
+  "amr": "acr:html-form:unitedstates",
+  "subdomain": "pdt-sample",
+  "region": "UnitedStates",
+  "sid": "Ldu2m7FqnP7cH6Gl"
+}
+```
+
+Now encrypting the payload using the algorithm specified in the header would give the following:
+
+```
+    eyJleHAiOjE2NDYwODYyNjAsIm5iZiI6MTY0NjA4MjY2MCwianRpIjoiYTlmZDQzYTQtYjAzNy00ZWViLTk4YjAtNDA1NDJlM2I5OGZmIiwiaXNzIjoiaHR0cHM6Ly9hcHAucGQtc3RhZ2luZy5jb20vZ2xvYmFsL29hdXRoL2Fub255bW91cyIsImF1ZCI6WyJodHRwczovL2FwaS5wZC1zdGFnaW5nLmNvbSIsIjkwYjE1M2JmLWNmZWItNDgzNy04YjZhLTg2NGI3ZmY4NTY1NiJdLCJzdWIiOiJmdGFudGF3aUBwYWdlcmR1dHkuY29tIiwiYXV0aF90aW1lIjoxNjQ2MDgyNjQ5LCJpYXQiOjE2NDYwODI2NjAsInB1cnBvc2UiOiJpZCIsImF0X2hhc2giOiJMTXRGS01MWWVGZXppTnR4OVBKV0hBIiwiYWNyIjoiYWNyOmh0bWwtZm9ybTp1bml0ZWRzdGF0ZXMiLCJkZWxlZ2F0aW9uX2lkIjoiNjU2NjAyN2QtNzcwNy00MjhmLTg3NTItM2VmOTcxYjc4ZDFhIiwiYWNjb3VudF9pZCI6IlA5SkVITUsiLCJ1c2VyX2lkIjoiUDhPR01JSyIsImF6cCI6IjkwYjE1M2JmLWNmZWItNDgzNy04YjZhLTg2NGI3ZmY4NTY1NiIsImFtciI6ImFjcjpodG1sLWZvcm06dW5pdGVkc3RhdGVzIiwic3ViZG9tYWluIjoicGR0LWZhcmVzIiwicmVnaW9uIjoiVW5pdGVkU3RhdGVzIiwic2lkIjoiTGR1Mm03RnFuUDdjSDZHbCJ9
+```
+
+### Signature
+
+The final part of the token is fairly different, as it isnt a full json object like the header or payload. Its made up of the encoded header and payload, again using the algorithm defined in the header, followed by a secret. The secret key value pair can be taken from the endpoint at https://api.pagerduty.com/global/oauth/anonymous/.well-known/openid-configuration which also contains information about the identity server. An example signature is the following:
```

Comment:
`https://app.pagerduty.com/global/oauth/anonymous/.well-known/openid-configuration`

Reply:
It might be simpler to link directly to https://app.pagerduty.com/global/oauth/anonymous/jwks, unless we think that might change

Reply:
true and it _shouldn't_ but partly why i wanna lean on giving them the former link since that should be the source of truth for everything

Reply:
I think it makes sense to give the jwks link. The openid-configuration link exposes a bunch of things that the user wouldn't have access to. For instance, the user-info url.